linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala

```diff
@@ -745,7 +709,16 @@ final class Emitter private (config: CommonPhaseConfig,
   }
 }
 
-private object Emitter {
+object Emitter {
+  /** Result of an emitter run. */
+  final class Result(
+      val body: List[js.Tree],
+      val header: String,
+      val footer: String,
```

Comment:
Nitpicking: consider ordering them as `header`, `body`, `footer`, to reflect the order in which they are supposed to be written to the file.